include/zephyr/ll_sched/ll_scheduler.h

```diff
@@ -0,0 +1,303 @@
+				   struct ll_sched_task *after);
+
+	/**
+	 * Sets task into running state along with executing additional
+	 * scheduler specific operations.
+	 * @param data Private data of selected scheduler.
+	 * @param task Task to be set into running state.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is optional.
+	 */
+	int (*schedule_task_running)(void *data, struct ll_sched_task *task);
+
+	/**
+	 * Sets task into completed state along with executing additional
+	 * scheduler specific operations.
+	 * @param data Private data of selected scheduler.
+	 * @param task Task to be set into completed state.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is optional.
+	 */
+	int (*schedule_task_complete)(void *data, struct ll_sched_task *task);
+
+	/**
+	 * Cancels previously scheduled task.
+	 * @param data Private data of selected scheduler.
+	 * @param task Task to be canceled.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is mandatory.
+	 */
+	int (*schedule_task_cancel)(void *data, struct ll_sched_task *task);
+
+	/**
+	 * Frees task's resources.
+	 *
+	 * This callout is responsible for freeing a task's resources. It shall
+	 * only be called from the context of a thread.
+	 *
+	 * @param data Private data of selected scheduler.
+	 * @param task Task to be freed.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is mandatory.
+	 */
+	int (*schedule_task_free)(void *data, struct ll_sched_task *task);
+
+	/**
+	 * Frees scheduler's resources.
+	 * @param data Private data of selected scheduler.
+	 * @param flags Function specific flags.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is optional.
+	 */
+	void (*scheduler_free)(void *data, uint32_t flags);
+
+	/**
+	 * Restores scheduler's resources.
+	 * @param data Private data of selected scheduler.
+	 * @return 0 if succeeded, error code otherwise.
+	 *
+	 * This operation is optional.
+	 */
+	int (*scheduler_restore)(void *data);
+};
+
+/** \brief Holds information about scheduler. */
+struct ll_sched {
+	sys_dnode_t  node;		/**< node in list of schedulers */
+	int type;			/**< SYS_SCHEDULE_ type */
+	const struct ll_sched_ops *ops;	/**< scheduler operations */
+	void *data;			/**< pointer to private data */
+};
+
+enum ll_sched_task_state {
+	LL_SCHED_TASK_STATE_INIT = 0,
+	LL_SCHED_TASK_STATE_QUEUED,
+	LL_SCHED_TASK_STATE_PENDING,
+	LL_SCHED_TASK_STATE_RUNNING,
+	LL_SCHED_TASK_STATE_PREEMPTED,
+	LL_SCHED_TASK_STATE_COMPLETED,
+	LL_SCHED_TASK_STATE_FREE,
+	LL_SCHED_TASK_STATE_CANCEL,
+	LL_SCHED_TASK_STATE_RESCHEDULE,
+};
+
+/** \brief Task operations. */
+struct ll_sched_task_ops {
+	enum ll_sched_task_state (*run)(void *data);	/**< task's main operation */
+	void (*complete)(void *data);		/**< executed on completion */
+	uint64_t (*get_deadline)(void *data);	/**< returns current deadline */
```

Comment:
first occurence, with several other incorrectly aligned comments further down: tab size is 8 characters in Zephyr, could you please fix comments alignment accordingly?